developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -1629,63 +1629,46 @@ end
 
 Plugins can store Foreman-wide settings either in the database or a
 config file. The DB should be preferred as it can be managed from the UI
-(under Administer > Settings), CLI, API and Puppet, and changed on the
+(under Administer > Settings), CLI and API. It also can be changed on the
 fly, while the config file is usually only used for settings that change
 behaviour during app startup and require a restart.
 
-To add new DB settings, the plugin should define a Setting class named
-after the plugin. For foreman_example, this would be at
-`app/models/setting/example.rb` and should look like:
+To add DB settings, the plugin should define them in it's registration block:
 
 [source, ruby]
 ....
-class Setting::Example < ::Setting
-  def self.load_defaults
-    return unless ActiveRecord::Base.connection.table_exists?('settings')
-    return unless super
-
-    Setting.transaction do
-      [
-        self.set('example_string', N_('Example setting that controls something'), 'default value'),
-        self.set('example_int', N_('Answer to the life, universe, and everything'), 42),
-      ].compact.each { |s| self.create s.update(:category => "Setting::Example") }
+Foreman::Plugin.register :my_plugin do
+  # ....
+
+  settings do
+    # Following settings will be added to a General category
+    category(:general) do
+      setting 'example_setting',
+        default: 'default value',
+        description: N_('Example setting that controls something'),
+        full_name: N_('Example of general setting')
```

Comment:
This may be just me, but I'd prefer the `full_name` before `description`. Descriptions can be long. Slight nit, but something to think about.

Reply:
That's my feeling as well, but we still have description required, but full_name can be ommited.
Though I agree that does not matter now and might be a good idea to require full_name, so :+1: